spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/utils/RewriteRowLevelOperationHelper.scala

```diff
@@ -103,7 +103,7 @@ trait RewriteRowLevelOperationHelper extends PredicateHelper with Logging {
 
     val scan = scanBuilder.asIceberg.withMetadataColumns(FILE_NAME_COL, ROW_POS_COL).build()
     val outputAttrs = toOutputAttrs(scan.readSchema(), tableAttrs)
-    val scanRelation = DataSourceV2ScanRelation(table, scan, outputAttrs)
+    val scanRelation = DataSourceV2ScanRelation(relation, scan, outputAttrs)
```

Comment:
This is our big 3.0 vs 3.1 change right? So here is where we could dynamically lookup the DSV2ScanRelation constructor